src/nn/ops/conv2d.ts

```diff
@@ -131,6 +131,65 @@ export class Conv2d extends SingleOutputOperation implements FusedOperation {
     return inputs;
   }
 
+  createOutput(): void {
+    let batches, channels, inputHeight, inputWidth, filterHeight, filterWidth;
+    this.tfPadding_ = utils.getPaddings(this.padding_);
+    const inputShape = this.input_.shape();
+
+    if (this.inputLayout_ === MLInputOperandLayout.nchw) {
+      [batches, , inputHeight, inputWidth] = inputShape;
+    } else {
+      // nhwc layout
+      [batches, inputHeight, inputWidth, ] = inputShape;
+    }
+
+    const filterShape = this.filter_.shape();
+    if (this.filterLayout_ === MLConv2dFilterOperandLayout.oihw) {
+      channels = filterShape[0];
+      filterHeight = filterShape[2];
+      filterWidth = filterShape[3];
+    } else if (this.filterLayout_ === MLConv2dFilterOperandLayout.hwio) {
+      channels = filterShape[3];
+      filterHeight = filterShape[0];
+      filterWidth = filterShape[1];
+    } else if (this.filterLayout_ === MLConv2dFilterOperandLayout.ihwo) {
+      channels = filterShape[3];
+      filterHeight = filterShape[1];
+      filterWidth = filterShape[2];
+    } else if (this.filterLayout_ === MLConv2dFilterOperandLayout.ohwi) {
+      channels = filterShape[0];
+      filterHeight = filterShape[1];
+      filterWidth = filterShape[2];
+    }
+
+    const effectiveFilterHeight =
+        filterHeight + (filterHeight - 1) * (this.dilations_[0] - 1);
+    const effectiveFilterWidth =
+        filterWidth + (filterWidth- 1) * (this.dilations_[1] - 1);
+
+    // output size = 1 +
+    //     (input size - filter size - (filter size - 1) * (dilation - 1) +
+    //      beginning padding + ending padding) / stride
+    const outputHeight =
+        1 + Math.floor((inputHeight - effectiveFilterHeight +
+          this.tfPadding_[1][0] + this.tfPadding_[1][1]) / this.strides_[0]);
+    const outputWidth =
+        1 + Math.floor((inputWidth - effectiveFilterWidth +
+          this.tfPadding_[2][0] + this.tfPadding_[2][1]) / this.strides_[1]);
+
+    // A depthwise conv2d operation is a variant of grouped convolution, used
+    // in models like the MobileNet, where the
+    //   options.groups = input_channels = output_channels
```

Comment:
Why do you need mention depthwise conv2d here? Are there any codes to handle it specifically?

Reply:
Thanks for gentle reminder. 
Yes, it would be confused with this comments, removed in new commit.